src/io/video.js

```diff
@@ -89,6 +64,14 @@ class Video {
         return 1;
     }
 
+    /**
+     * Set a video provider for this device.
+     * @param {VideoProvider} provider - Video provider to use
+     */
+    setProvider (provider) {
+        this.provider = provider;
```

Comment:
Do you think we might want to track enabled/disabled state here and enable if the provider is added late?

I don't think we should worry to much about "state fixing" in the setProvider call - the intention is that a provider would be set at creation time, and not changed while running.